File: examples/js/01_minimal_chat/main.ts
/** Smallest possible ai-agent run in TypeScript. */

import { fileURLToPath } from 'node:url';
import { dirname, resolve } from 'node:path';

import { Agent } from '@ai-agent/sdk';

// Resolve ../../../agent relative to this file (examples/js/<name>/main.ts).
const HERE = dirname(fileURLToPath(import.meta.url));
const AGENT_BINARY = resolve(HERE, '../../../agent');

const env = {
  SLLM_ENDPOINT: process.env.SLLM_ENDPOINT ?? 'http://localhost:8080/v1/chat/completions',
  SLLM_API_KEY: process.env.SLLM_API_KEY ?? 'sk-gemma4',
};

async function main(): Promise<void> {
  // `await using` ensures close() runs even on error.
  await using agent = await Agent.open({ binaryPath: AGENT_BINARY, env });
  const result = await agent.run('こんにちは。今日の天気はどう？一文で答えて。');
  console.log('response:', result.response);
  console.log('reason:  ', result.reason);
  console.log('turns:   ', result.turns);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: examples/python/01_minimal_chat/main.py
"""Smallest possible ai-agent run: spawn the binary, ask one question, exit."""

from __future__ import annotations

import asyncio
import os
from pathlib import Path

from ai_agent import Agent

# Resolve the agent binary built at the repo root: examples/python/<name>/main.py
AGENT_BINARY = str(Path(__file__).resolve().parents[3] / "agent")

# SLLM endpoint and API key — defaults match the repo's reference LM Studio setup.
ENV = {
    "SLLM_ENDPOINT": os.getenv("SLLM_ENDPOINT", "http://localhost:8080/v1/chat/completions"),
    "SLLM_API_KEY": os.getenv("SLLM_API_KEY", "sk-gemma4"),
}


async def main() -> None:
    # Async context manager handles start() / close() for us.
    async with Agent(binary_path=AGENT_BINARY, env=ENV) as agent:
        result = await agent.run("こんにちは。今日の天気はどう？一文で答えて。")
        print("response:", result.response)
        print("reason:  ", result.reason)
        print("turns:   ", result.turns)


if __name__ == "__main__":
    asyncio.run(main())
